src/platform/Tizen/ConnectivityUtils.cpp

```diff
@@ -72,14 +98,602 @@ app::Clusters::GeneralDiagnostics::InterfaceTypeEnum ConnectivityUtils::GetInter
+
+                memcpy(ifp->Ipv4AddressesBuffer[index], addPtr, kMaxIPv4AddrSize);
+                ifp->Ipv4AddressSpans[index] = ByteSpan(ifp->Ipv4AddressesBuffer[index], kMaxIPv4AddrSize);
+                index++;
+
+                if (index >= kMaxIPv4AddrCount)
+                {
+                    break;
+                }
+            }
+        }
+    }
+
+    if (index > 0)
+    {
+        err  = CHIP_NO_ERROR;
+        size = index;
+    }
+
+    freeifaddrs(ifaddr);
+    return err;
+}
+
+CHIP_ERROR ConnectivityUtils::GetInterfaceIPv6Addrs(const char * ifname, uint8_t & size, NetworkInterface * ifp)
+{
+    CHIP_ERROR err          = CHIP_ERROR_READ_FAILED;
+    struct ifaddrs * ifaddr = nullptr;
+
+    if (getifaddrs(&ifaddr) == -1)
+    {
+        ChipLogError(DeviceLayer, "Failed to get network interfaces");
+        return err;
+    }
+
+    uint8_t index = 0;
+    for (struct ifaddrs * ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next)
+    {
+        if (ifa->ifa_addr && ifa->ifa_addr->sa_family == AF_INET6)
+        {
+            if (strcmp(ifname, ifa->ifa_name) == 0)
+            {
+                void * addPtr = &((struct sockaddr_in6 *) ifa->ifa_addr)->sin6_addr;
+
+                memcpy(ifp->Ipv6AddressesBuffer[index], addPtr, kMaxIPv6AddrSize);
+                ifp->Ipv6AddressSpans[index] = ByteSpan(ifp->Ipv6AddressesBuffer[index], kMaxIPv6AddrSize);
+                index++;
+
+                if (index >= kMaxIPv6AddrCount)
+                {
+                    break;
+                }
+            }
+        }
+    }
+
+    if (index > 0)
+    {
+        err  = CHIP_NO_ERROR;
+        size = index;
+    }
+
+    freeifaddrs(ifaddr);
+    return err;
+}
+
+CHIP_ERROR ConnectivityUtils::GetWiFiInterfaceName(char * ifname, size_t bufSize)
+{
+    CHIP_ERROR err          = CHIP_ERROR_READ_FAILED;
+    struct ifaddrs * ifaddr = nullptr;
+
+    if (getifaddrs(&ifaddr) == -1)
+    {
+        ChipLogError(DeviceLayer, "Failed to get network interfaces");
+        return err;
+    }
+
+    struct ifaddrs * ifa = nullptr;
+    for (ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next)
+    {
+        if (GetInterfaceConnectionType(ifa->ifa_name) == InterfaceTypeEnum::EMBER_ZCL_INTERFACE_TYPE_ENUM_WI_FI)
+        {
+            Platform::CopyString(ifname, bufSize, ifa->ifa_name);
+            err = CHIP_NO_ERROR;
+            break;
+        }
+    }
+
+    freeifaddrs(ifaddr);
+    return err;
+}
+
+CHIP_ERROR ConnectivityUtils::GetWiFiChannelNumber(const char * ifname, uint16_t & channelNumber)
+{
+    CHIP_ERROR err = CHIP_ERROR_READ_FAILED;
+
+    struct iwreq wrq;
+    int skfd;
+
+    if ((skfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0)
+    {
+        ChipLogError(DeviceLayer, "Failed to create a channel to the NET kernel.");
+        return CHIP_ERROR_OPEN_FAILED;
+    }
+
+    if (GetWiFiParameter(skfd, ifname, SIOCGIWFREQ, &wrq) == CHIP_NO_ERROR)
```

Comment:
How about forwarding error code to the caller?
```suggestion
    if ((err = GetWiFiParameter(skfd, ifname, SIOCGIWFREQ, &wrq)) == CHIP_NO_ERROR)
```

Same for `GetWiFiStats` in functions below.

Reply:
Thank you for your good point. i'll update it.